### DISCUSSION TITLE
Compound data in chart with previous values

### DISCUSSION THREAD

**Discussion by Kuuchuu** [Budibase/budibase#3678] [Help]
Hello! Is there a way for me to set my line chart to compound all values before it so that it shows the sum of all data available at a certain time?
![BudibaseChart](https://user-images.githubusercontent.com/20036947/144726299-2d1289d9-129c-4ad2-898c-5cd8645a9512.png)
Thanks!

**Comment by mjashanks** [comment]
Hi @Kuuchuu 

Normally the answer for this would be to use a view, with a Group By and calculation: https://docs.budibase.com/building-apps/data/views

Unfortunately, I think that you would need an additional step. I see that you have exact datetimes. If you wish to compound values per day (for example), you would need to write a formula to convert `2021-11-20T20:51:46Z` to `2021-11-20`. Whilst you can write formula fields in Budibase, you cannot group by formula inside view.

So, it's not currently possible to do what you need. Here are list of limitations of views, which we hope to addrees sometime: https://github.com/Budibase/budibase/discussions/2999. 

I have linked this discussion to that #2999.

**Reply by Kuuchuu** [reply]
@mjashanks  gotcha, I can modify the data the database itself receives so that it is YYYY-MM-DD--HH:MM, or something similar. There may be a bunch of sales during the same minute and they would want to see that, so I may try grouping them by minute. I'm assuming the "view" would be capable of working with minutes? Thanks for the help

**Reply by Kuuchuu** [reply]
Is it not possible to create a view with a REST query? I did see you can create a view with an internal database. I can use that if needed. I've found how to find the sum of the values, but it only adds the values to others in the same group, and doesn't combine with other values before it. I'm assuming that isn't built in quite yet? I'm assuming the route I'll have to take is creating a JavaScript function to transform the query so that it is adding as it goes?

**Reply by mjashanks** [reply]
REST queries have transformers: https://docs.budibase.com/building-apps/data/transformers
